packages/iTwinUI-react/src/core/ComboBox/ComboBox.tsx

```diff
@@ -153,32 +153,46 @@ export const ComboBox = <T,>(props: ComboBoxProps<T>) => {
   const toggleButtonRef = React.useRef<HTMLSpanElement>(null);
   const mounted = React.useRef(false);
 
+  // Record to store all extra information (e.g. original indexes), where the key is the id of the option
```

Comment:
Does the order where this bit of code appears matter in the file? (curious)

Reply:
Computer reads code from top to bottom so order where this bit of code appears in the file depends where I'm using it. 
To use variable/function/etc., it firstly needs to be declared.
I created a function (to reuse code) `initializeExtraInfo` which uses `optionsExtraInfoRef`. Therefore `optionsExtraInfoRef` needs to be above said function.